lib/matplotlib/font_manager.py

```diff
@@ -1177,25 +1177,39 @@ def score_size(self, size1, size2):
     def findfont(self, prop, fontext='ttf', directory=None,
                  fallback_to_default=True, rebuild_if_missing=True):
         """
-        Search the font list for the font that most closely matches
-        the :class:`FontProperties` *prop*.
-
-        :meth:`findfont` performs a nearest neighbor search.  Each
-        font is given a similarity score to the target font
-        properties.  The first font with the highest score is
-        returned.  If no matches below a certain threshold are found,
-        the default font (usually DejaVu Sans) is returned.
-
-        `directory`, is specified, will only return fonts from the
-        given directory (or subdirectory of that directory).
+        Find a font that most closely matches the given font properties.
+
+        Parameters
+        ----------
+        prop : `~matplotlib.fontmanager.FontProperties`
+            The font properties to search for.
+        fontext : str, optional, default: 'ttf'
+            The extension of font file; e.g. 'ttf' or 'afm'.
+        directory : str, optional
+            If given, only search this directory and its subdirectories.
+        fallback_to_default : bool
+            If True, will fallback to the default font family (usually
+            "DejaVu Sans" or "Helvetica") if the first lookup hard-fails.
```

Comment:
I don't quite understand the behavior for `fallback_to_default=False`. This also returns "DejaVu Sans". Not sure if this is a bug. Anyway, we're missing documentation for the non-fallback result.

Reply:
In one case we still try to match the font size, weight, and style (italic, bold, etc.) but not in the other.  TBH I doubt the distinction really matters and would suggest just deprecating that argument (can be done in a separate PR, of course).

Reply:
Ok. I'll take care of that later in a separate PR.